consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/blockcreation/MergeCoordinator.java

```diff
@@ -310,6 +323,44 @@ public ForkchoiceResult updateForkChoice(
     return ForkchoiceResult.withResult(newFinalized, Optional.of(newHead));
   }
 
+  private boolean setNewHead(final MutableBlockchain blockchain, final BlockHeader newHead) {
+
+    if (newHead.getHash().equals(blockchain.getChainHeadHash())) {
+      debugLambda(LOG, "Nothing to do new head {} is already chain head", newHead::toLogString);
+      return true;
+    }
+
+    if (newHead.getParentHash().equals(blockchain.getChainHeadHash())) {
+      debugLambda(
+          LOG,
+          "Forwarding chain head to the block {} saved from a previous newPayload invocation",
+          newHead::toLogString);
+      forwardWorldStateTo(newHead);
```

Comment:
it might be worthwhile to see if this represents an append to the chain/worldstate rather than a reorg.  A reorg to a prior or peer block probably should use rewind to correctly handle transactions and receipts.